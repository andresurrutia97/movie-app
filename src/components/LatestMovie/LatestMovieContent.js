import React from 'react';
import PropTypes from 'prop-types';

const Content = ({ children }) => {
  return (
    <div className="absolute flex flex-col justify-center items-center top-0 mt-3 w-full md:w-4/6 md:items-start md:pt-24 md:pl-32 lg:pt-24 lg:pl-40 xl:mt-40 xl:pl-56">
      {children}
    </div>
  );
};

Content.propTypes = {
  children: PropTypes.node,
};

Content.defaultProps = {
  children: null,
};

export default Content;
