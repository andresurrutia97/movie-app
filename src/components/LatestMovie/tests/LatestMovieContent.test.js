import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import LatestMovieContent from '../LatestMovieContent';

describe('<Content/>', () => {
  const component = (
    <LatestMovieContent>
      <div>Test</div>
    </LatestMovieContent>
  );

  it('should render correctly ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should coantain a <div/> as a children', () => {
    const wrapper = shallow(component);
    expect(wrapper.children().find('div')).toHaveLength(1);
    expect(wrapper.children().find('div').props().children).toBe('Test');
  });
});
