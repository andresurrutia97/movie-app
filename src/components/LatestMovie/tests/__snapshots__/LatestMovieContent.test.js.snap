// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content/> should render correctly  1`] = `
<div
  className="absolute flex flex-col justify-center items-center top-0 mt-3 w-full md:w-4/6 md:items-start md:pt-24 md:pl-32 lg:pt-24 lg:pl-40 xl:mt-40 xl:pl-56"
>
  <div>
    Test
  </div>
</div>
`;
